--- lame-3.97/libmp3lame/Makefile.am.no_undefined	2005-08-21 12:32:09.000000000 -0500
+++ lame-3.97/libmp3lame/Makefile.am	2006-10-20 12:06:51.000000000 -0500
@@ -18,7 +18,7 @@
 decoder_ldadd =
 endif
 
-libmp3lame_la_LIBADD = $(nasm_ldadd) $(decoder_ldadd)
+libmp3lame_la_LIBADD = $(nasm_ldadd) $(decoder_ldadd) @CONFIG_MATH_LIB@
 libmp3lame_la_LDFLAGS = -version-info @LIB_MAJOR_VERSION@:@LIB_MINOR_VERSION@ \
 			-no-undefined
 
--- lame-3.97/configure.in.no_undefined	2006-09-24 08:53:31.000000000 -0500
+++ lame-3.97/configure.in	2006-10-20 12:07:15.000000000 -0500
@@ -1059,8 +1059,9 @@
 
 CFLAGS="${OPTIMIZATION} ${CFLAGS}"
 LDADD="${LDADD}"
-FRONTEND_LDADD="${FRONTEND_LDADD} ${CONFIG_MATH_LIB}"
+FRONTEND_LDADD="${FRONTEND_LDADD}"
 
+AC_SUBST(CONFIG_MATH_LIB)
 
 AC_SUBST(INCLUDES)
 
